wlauto/workloads/iozone/__init__.py

```diff
@@ -0,0 +1,136 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from wlauto import Workload, Parameter, Executable
+from wlauto.exceptions import ConfigError
+from wlauto.utils.types import list_of_strs
+import os
+import re
+
+iozone_results_txt = 'iozone_results.txt'
+
+time_res_regex = re.compile(r'Time Resolution = (\d+.\d+) (\w+)')
+cache_size_regex = re.compile(r'Processor cache size set to (\d+) (\w+)')
+cache_line_regex = re.compile(r'Processor cache line size set to (\d+) (\w+)')
+
+
+class Iozone(Workload):
+
+    name = 'iozone'
+    description = """
+    Iozone runs a series of disk I/O performance tests.
+
+    To run specific tests, they must be written in the form of:
+
+    ['0', '1', '5', '6']
+
+    By default, iozone will run all tests in auto mode.
+
+    The official website for iozone is at www.iozone.org.
+    """
+
+    parameters = [
+        Parameter('tests', kind=list_of_strs, default=[],
+                  description='List of performance tests to run.'),
+        Parameter('auto_mode', kind=bool, default=True,
+                  description='Run tests in auto mode.'),
+        Parameter('timeout', kind=int, default=14400,
+                  description='Timeout for the workload.'),
+        Parameter('file_size', kind=int, default=0,
+                  description='Fixed file size to run tests with.'),
+        Parameter('record_length', kind=int, default=0,
+                  description='Fixed record length.'),
+        Parameter('threads', kind=int, default=0,
+                  description='Number of threads'),
+        Parameter('other_params', kind=str, default='',
+                  description='Other parameter. Run iozone -h to see'
+                              ' list of options.')
+    ]
+
+    def initialize(self, context):
+        Iozone.host_binary = context.resolver.get(Executable(self,
+                                                             self.device.abi,
+                                                             'iozone'))
+        Iozone.device_binary = self.device.install(Iozone.host_binary)
+
+    def setup(self, context):
+        self.results = os.path.join(self.device.working_directory,
+                                    iozone_results_txt)
+        self.command = self._build_command()
+
+        if self.threads and self.auto_mode:
+            raise ConfigError("You cannot set the number of threads and enable"
+                              " auto mode at the same time.")
+
+    def _build_command(self):
+        iozone_command = '{}'.format(self.device_binary)
```

Comment:
iozone creates a temporary file in in the current working directory, which may not be writable. This should therefore be:

     iozone_command = 'cd {} && {}'.format(self.device.working_directory, self.device_binary)